man/systemd-system.conf.xml

```diff
@@ -128,6 +128,7 @@
       <varlistentry>
         <term><varname>RuntimeWatchdogSec=</varname></term>
         <term><varname>ShutdownWatchdogSec=</varname></term>
+        <term><varname>ShutdownWatchdogOnKexec=</varname></term>
```

Comment:
hmm, instead of declaring this as boolean I think it should be a time value of its own. i.e. this should probably be called `KexecWatchdogSec=` and should define a timeout that is configurable independently of ShutdownWatchdogSec=, and unlike it would default to off. I think this would make a lot of sense, given that kexec is much faster than regular reboots typically, and thus a setting a different setup would make a ton of sense in many cases.

(And quite frankly, we should probably rename ShutdownWatchdogSec= to RebootWatchdogSec= since it it's only applied for reboots, not the other cases of shutdown, i.e. poweroff, halt or kexec, hapy to take a patch for that too — which should still support the old name for compat reasons but remove it from the docs)

Reply:
> hmm, instead of declaring this as boolean I think it should be a time value of its own. i.e. this should probably be called KexecWatchdogSec= and should define a timeout that is configurable independently of ShutdownWatchdogSec=, and unlike it would default to off. I think this would make a lot of sense, given that kexec is much faster than regular reboots typically, and thus a setting a different setup would make a ton of sense in many cases.

Sure, that makes sense - it was the other option proposed in #13111

> (And quite frankly, we should probably rename ShutdownWatchdogSec= to RebootWatchdogSec= since it it's only applied for reboots, not the other cases of shutdown, i.e. poweroff, halt or kexec, hapy to take a patch for that too — which should still support the old name for compat reasons but remove it from the docs)

Sounds like a good change to make - I'll test both changes tomorrow morning and once they work fine will push an update to this PR.

Thanks for the feedback!

Reply:
All done, seems to work fine on my hardware